stdlib/http/src/main/ballerina/http/resiliency/http_circuit_breaker.bal

```diff
@@ -505,7 +505,14 @@ function updateCircuitHealthFailure(CircuitHealth circuitHealth,
         bucket.failureCount += 1;
         time:Time lastUpdated = time:currentTime();
         circuitHealth.lastErrorTime = lastUpdated;
-        circuitHealth.totalBuckets[currentBucketId].lastUpdatedTime = lastUpdated;
+        Bucket?[] buckets = circuitHealth.totalBuckets;
+        if (buckets is Bucket[]) {
```

Comment:
Is this check correct?

This will only be true if `buckets` is frozen right? 

Shouldn't this be something like 
```ballerina
    Bucket?[] buckets = circuitHealth.totalBuckets;
    foreach var nillableBucket in buckets {
        time:Time? lastUpdatedTime = nillableBucket?.lastUpdatedTime;
    }
```